lstchain/scripts/lstchain_r0g_to_r0v.py

```diff
@@ -0,0 +1,175 @@
+    except (AttributeError, IndexError):
+        log.error('CameraConfiguration not found! Is this R1v1 data?')
+        sys.exit(1)
+
+    num_pixels = camera_config.num_pixels
+    num_samples = camera_config.num_samples_nominal
+    pixel_id_map = protozfits.any_array_to_numpy(camera_config.pixel_id_map)
+
+    num_events = 0
+
+    with ExitStack() as stack:
+        for i, name in enumerate(output_stream_names):
+            header = input_streams[i].Events.header
+            n_tiles = header["NAXIS2"]
+            rows_per_tile = header["ZTILELEN"]
+
+            stream = stack.enter_context(protozfits.ProtobufZOFits(
+                    n_tiles=n_tiles,
+                    rows_per_tile=rows_per_tile,
+                    compression_block_size_kb=64*1024,
+                    defaul_compression="zstd9"))
+            stream.open(str(name))
+
+            stream.move_to_new_table("DataStream")
+            stream.write_message(input_streams[i].DataStream[0])
+          
+            stream.move_to_new_table("CameraConfiguration")
+            stream.write_message(input_streams[i].CameraConfiguration[0])
+
+            stream.move_to_new_table("Events")
+
+            for event in input_streams[i].Events:
+                # skip corrupted events:
+                if event.event_id == 0:
+                    continue
+                num_events += 1
+
+                # Check if this event is known
+                if event.event_id not in pixel_mask:
+                    log.warning('Event id %d not found in pixel selection file!',
+                             event.event_id)
+                    continue
+
+                num_gains = event.num_channels
+                wf = protozfits.any_array_to_numpy(event.waveform)
+                wf = wf.reshape((num_gains, num_pixels, num_samples))
+
+                evtype = event_type[event.event_id]
+
+                if evtype in EVENT_TYPES_TO_REDUCE:
+                    pixmask = pixel_mask[event.event_id]
+                    ordered_pix_mask = pixmask[pixel_id_map]
+                    new_wf = wf[:, ordered_pix_mask, :]
+                    event.waveform.data = new_wf.tobytes()
+                    pixel_status = protozfits.any_array_to_numpy(event.pixel_status)
+                    # Modify pixel status as needed
+                    new_status = np.where(ordered_pix_mask,
+                                          pixel_status | PixelStatus.DVR_STATUS_0,
+                                          pixel_status)
+                    event.pixel_status.data = new_status.tobytes()
+
+                stream.write_message(event)
+
+            stream.close()
+            input_streams[i].close()
+
+            # Open again output file for setting the LSTDVR flag:
+            outfile = fits.open(name, mode="update")
+            outfile['Events'].header.set("LSTDVR", True, 
+                                         "LST offline DVR applied")
+            outfile.close()
```

Comment:
Not that it changes much, but can't this be set already above, while you have the file opened?

Reply:
No, it can't be done like that... protzfits does not have the functions to do it.

Reply:
I just released protozfits 2.5 with added `set_{str,bool,int,float}` methods on `ProtobufZOFits` and also the possibility to set the per-column compressions. 

The header keywords need to be set before the first row of the table is written.

Reply:
If using astropy, you should use a context manager:

```
with fits.open(name, mode="update") as outfile:
    outfile["Events"].header.set("LSTDVR", True, "LST offline DVR applied")
```

Reply:
I have to change the protozfits version in environment.yml, right?  (not needed in setup.py which has >=2.2)

Reply:
Since we would be using features only available with 2.5, we need to update the requirements to be `>= 2.5, <3`